htlcswitch/switch.go

```diff
@@ -1322,16 +1322,24 @@ func (s *Switch) handlePacketForward(packet *htlcPacket) error {
 					packet.incomingChanID, packet.incomingHTLCID,
 					packet.outgoingChanID, packet.outgoingHTLCID)
 
-				fail.Reason = circuit.ErrorEncrypter.EncryptMalformedError(
-					fail.Reason,
-				)
+				fail.Reason, err = circuit.ErrorEncrypter.
+					EncryptMalformedError(fail.Reason)
+				if err != nil {
+					err = fmt.Errorf("unable to obfuscate "+
+						"malformed error: %v", err)
+					log.Error(err)
+				}
 
 			default:
 				// Otherwise, it's a forwarded error, so we'll perform a
 				// wrapper encryption as normal.
-				fail.Reason = circuit.ErrorEncrypter.IntermediateEncrypt(
-					fail.Reason,
-				)
+				fail.Reason, err = circuit.ErrorEncrypter.
+					IntermediateEncrypt(fail.Reason)
+				if err != nil {
+					err = fmt.Errorf("unable to obfuscate "+
```

Comment:
Todo: create a proper failure message to attribute blame to the downstream peer. See https://github.com/lightningnetwork/lightning-onion/pull/60#discussion_r1228035385

Reply:
Added correct handling when an intermediate node receives an invalid failure message from downstream.